hw/xbox/xid-sdl.c

```diff
@@ -511,18 +557,11 @@ static void usb_xbox_gamepad_realize(USBDevice *dev, Error **errp)
     SDL_JoystickEventState(SDL_ENABLE);
 #endif
 
-#ifdef FORCE_FEEDBACK
-    s->sdl_haptic = SDL_HapticOpenFromJoystick(s->sdl_gamepad);
-    if (s->sdl_haptic == NULL) {
-        fprintf(stderr, "Joystick doesn't support haptic\n");
-    } else {
-        if ((SDL_HapticQuery(s->sdl_haptic) & SDL_HAPTIC_LEFTRIGHT) == 0) {
-            fprintf(stderr, "Joystick doesn't support necessary haptic effect\n");
-            SDL_HapticClose(s->sdl_haptic);
-            s->sdl_haptic = NULL;
-        }
+    if (SDL_InitSubSystem(SDL_INIT_HAPTIC)) {
+        fprintf(stderr, "SDL failed to initialize haptic feedback subsystem\n");
     }
```

Comment:
This should probably be in the class-init, so it's only called once. Not for every instance.
The same is true for the other SDL init though.

We should probably create an issue to review these things and do a follow-up PR, and refer developers to https://people.redhat.com/~thuth/blog/qemu/2018/09/10/instance-init-realize.html

Reply:
Good call, created ticket here: https://github.com/xqemu/xqemu/issues/210